runtime/src/snapshot_controller.rs

```diff
@@ -0,0 +1,193 @@
+use {
+    crate::{
+        accounts_background_service::{AbsRequestSender, SnapshotRequest, SnapshotRequestKind},
+        bank::{epoch_accounts_hash_utils, Bank, SquashTiming},
+        bank_forks::SetRootError,
+        snapshot_config::SnapshotConfig,
+    },
+    log::*,
+    solana_measure::measure::Measure,
+    solana_sdk::clock::Slot,
+    std::{
+        cmp,
+        sync::{
+            atomic::{AtomicU64, Ordering},
+            Arc,
+        },
+        time::Instant,
+    },
+};
+
+#[derive(Default)]
+pub struct SnapshotController {
+    abs_request_sender: AbsRequestSender,
+    snapshot_config: Option<SnapshotConfig>,
+    latest_abs_request_slot: AtomicU64,
```

Comment:
The code in this new module is kept as similar as possible to the code in `BankForks` with a small exception that this is now an atomic to allow using `SnapshotController` without a mut access